mypy/checkexpr.py

```diff
@@ -675,6 +679,34 @@ def try_infer_partial_value_type_from_call(
 
         return None
 
+    def apply_function_signature_hook(
```

Comment:
Would it be possible to merge this with apply_method_signature_hook? Maybe with a joint helper that takes formal args, callee, context, and the checker, and then pass different versions of that to the joint helper?

Reply:
Actually I just did this myself